packages/zoe/test/unitTests/contractSupport/test-zoeHelpers.js

```diff
@@ -480,174 +302,192 @@ test.skip('ZoeHelpers isOfferSafe', t => {
+    assert(!exited, details`seat has been exited`);
+
+  let zoeSeatAdminExitCalled = false;
+  let zoeSeatAdminKickoutCalled = false;
+  const mockZoeSeatAdmin = harden({
+    exit: () => zoeSeatAdminExitCalled = true,
+    kickOut: () => zoeSeatAdminKickoutCalled = true,
+  });
+  const { zcfSeat: actual } = makeZcfSeatAdminKit(
+    allSeatStagings,
+    mockZoeSeatAdmin,
+    { proposal, initialAllocation},
+    getAmountMath,
+  );
+  const mockSeat = harden({
+    isOfferSafe: actual.isOfferSafe,
+    getCurrentAllocation: actual.getCurrentAllocation,
+    getProposal: () => proposal,
+    stage: actual.stage,
+    hasExited: actual.hasExited,
+  });
+  return mockSeat;
+}
+
+test('ZoeHelpers trade ok', t => {
+  const { moolaR, simoleanR, moola, simoleans, amountMaths, brands } = setup();
+  const getAmountMath =
+      brand => amountMaths.get(brand.getAllegedName());
+  const leftProposal = {
+    give: { Asset: moola(10) },
+    want: { Bid: simoleans(4) },
+    exit: { onDemand: null },
+  }
+  const leftAlloc = { Asset: moola(10) };
+  const leftZcfSeat = makeMockZcfSeatAdmin(leftProposal, leftAlloc, getAmountMath);
+  const rightProposal = {
+    give: { Money: simoleans(6) },
+    want: { Items: moola(7) },
+    exit: { onDemand: null },
+  }
+
+  const rightAlloc = { Money: simoleans(6) };
+  const rightZcfSeat = makeMockZcfSeatAdmin(rightProposal, rightAlloc, getAmountMath);
+  const mockZCFBuilder = makeMockTradingZcfBuilder();
+  mockZCFBuilder.addBrand(moolaR);
+  mockZCFBuilder.addBrand(simoleanR);
+  const mockZCF = mockZCFBuilder.build();
+  t.notThrows(
+    () =>
       trade(
+        mockZCF,
         {
-          offerHandle: leftOfferHandle,
+          seat: leftZcfSeat,
           gains: { Bid: simoleans(4) },
           losses: { Asset: moola(7) },
         },
         {
-          offerHandle: rightOfferHandle,
+          seat: rightZcfSeat,
           gains: { Items: moola(7) },
           losses: { Money: simoleans(4) },
         },
-      ),
-    );
-    t.deepEqual(
-      mockZCF.getReallocatedHandles(),
-      harden([leftOfferHandle, rightOfferHandle]),
-      `both handles reallocated`,
-    );
-    t.deepEqual(
-      mockZCF.getReallocatedAmountObjs(),
-      [
-        { Asset: moola(3), Bid: simoleans(4) },
-        { Money: simoleans(2), Items: moola(7) },
-      ],
-      `amounts reallocated passed to reallocate were as expected`,
-    );
-    t.deepEqual(
-      mockZCF.getCompletedHandles(),
-      harden([]),
-      `no handles were completed`,
-    );
+      ));
+  t.deepEqual(mockZCF.getReallocatedStagings().length, 2, `both reallocated`);
+  t.deepEqual(
+    mockZCF.getReallocatedStagings()[0].getStagedAllocation(),
+    { Asset: moola(3), Bid: simoleans(4) },
+    'left gets what she wants',
+  );
+  t.deepEqual(
+    mockZCF.getReallocatedStagings()[1].getStagedAllocation(),
+    { Items: moola(7), Money: simoleans(2) },
+    'right gets what he wants',
+  );
```

Comment:
We lost a test of whether the seats were exited or not. Can we confirm that both are still active? hasExited === false?

Reply:
I'm confused about what you think might happen.
This is a unit test of zoeHelpers.trade(). The seat is a mock. Since trade() doesn't call exit, the mock doesn't even have a method to be exited.  
swap() might call exit(), but we're not testing that here.

Reply:
> This is a unit test of zoeHelpers.trade(). The seat is a mock. 

You don't say? ;) 

I think what I'm suggesting is very easy to implement. As the old tests did, we should record whether the zcfSeat.exit() gets called or not, and in this particular test, assert that trade does not ever call `zcfSeat.exit` on the seats which are arguments.

Reply:
Since the mock seats don't implement exit(), the test will throw an error and fail if exit is called. I'm not sure that adding mechanisms to the mock to detect that will make it more robust.

Reply:
Surely the mockSeats can implement a mock for exit, though. The point is to test that exit is not called.

Reply:
Yes, they can. The test and the mocks seem less robust to me this way.   

Since these mock seats are explicitly intended for testing `trade()` they don't expect to have `exit()` called, and would throw an exception. Now they have an exit method, and every test that uses the test needs to verify that it wasn't called.

Reply:
Ah, ok. So calling exit() would already throw an exception? That sounds like it performs the test I was envisioning.